Sources/SwiftTUI/RunLoop/Application.swift

```diff
@@ -61,41 +63,46 @@ public class Application {
     }
 
     private func handleInput() {
-        var char: UInt8 = 0
-        _ = read(STDIN_FILENO, &char, 1)
-
-        if arrowKeyParser.parse(character: char) {
-            guard let key = arrowKeyParser.arrowKey else { return }
-            arrowKeyParser.arrowKey = nil
-            if key == .down {
-                if let next = window.firstResponder?.selectableElement(below: 0) {
-                    window.firstResponder?.resignFirstResponder()
-                    window.firstResponder = next
-                    window.firstResponder?.becomeFirstResponder()
-                }
-            } else if key == .up {
-                if let next = window.firstResponder?.selectableElement(above: 0) {
-                    window.firstResponder?.resignFirstResponder()
-                    window.firstResponder = next
-                    window.firstResponder?.becomeFirstResponder()
-                }
-            } else if key == .right {
-                if let next = window.firstResponder?.selectableElement(rightOf: 0) {
-                    window.firstResponder?.resignFirstResponder()
-                    window.firstResponder = next
-                    window.firstResponder?.becomeFirstResponder()
-                }
-            } else if key == .left {
-                if let next = window.firstResponder?.selectableElement(leftOf: 0) {
-                    window.firstResponder?.resignFirstResponder()
-                    window.firstResponder = next
-                    window.firstResponder?.becomeFirstResponder()
+        let data = FileHandle.standardInput.availableData
+
+        guard let string = String(data: data, encoding: .utf8) else {
```

Comment:
I wonder if there is any chance the `availableData` from the stream is cut right in the middle of a character, returning the remaining bytes only on the next run loop cycle. Seems to work fine, though.

Reply:
That's a good question... I'm inclined to believe that this is not an issue with the Terminal app (after testing it by both typing and copy-pasting stuff in), but I have an acquaintance well-versed in the internals of macOS, I'll ask her for her opinion.

Reply:
Yes, that'd be great! I'll merge it anyway because it's a big improvement over what it was before already.